neuralop/layers/mwno_block.py

```diff
@@ -0,0 +1,701 @@
+        PHI0 : ndarray of shape (k, k)
+            Reconstruction matrix for even samples
+        PHI1 : ndarray of shape (k, k)
+            Reconstruction matrix for odd samples
+            
+        Notes
+        -----
+        The filters satisfy the two-scale relations:
+        - φ(x) = √2 * Σ H[n] * φ(2x - n)
+        - ψ(x) = √2 * Σ G[n] * φ(2x - n)
+        
+        Where H are low-pass (scaling) filters and G are high-pass (wavelet) filters.
+        """
+        def psi(psi1, psi2, i, inp):
+            mask = (inp<=0.5) * 1.0
+            return psi1[i](inp) * mask + psi2[i](inp) * (1-mask)
+        
+        if base not in ['legendre', 'chebyshev']:
+            raise Exception('Base not supported')
+        
+        x = Symbol('x')
+        H0 = np.zeros((k,k))
+        H1 = np.zeros((k,k))
+        G0 = np.zeros((k,k))
+        G1 = np.zeros((k,k))
+        PHI0 = np.zeros((k,k))
+        PHI1 = np.zeros((k,k))
+        
+        phi, psi1, psi2 = cls.get_phi_psi(k, base)
+        
+        if base == 'legendre':
+            roots = Poly(legendre(k, 2*x-1)).all_roots()
+            x_m = np.array([rt.evalf(20) for rt in roots]).astype(np.float64)
+            wm = 1/k/cls.legendreDer(k,2*x_m-1)/eval_legendre(k-1,2*x_m-1)
+            
+            for ki in range(k):
+                for kpi in range(k):
+                    H0[ki, kpi] = 1/np.sqrt(2) * (wm * phi[ki](x_m/2) * phi[kpi](x_m)).sum()
+                    G0[ki, kpi] = 1/np.sqrt(2) * (wm * psi(psi1, psi2, ki, x_m/2) * phi[kpi](x_m)).sum()
+                    H1[ki, kpi] = 1/np.sqrt(2) * (wm * phi[ki]((x_m+1)/2) * phi[kpi](x_m)).sum()
+                    G1[ki, kpi] = 1/np.sqrt(2) * (wm * psi(psi1, psi2, ki, (x_m+1)/2) * phi[kpi](x_m)).sum()
+                    
+            PHI0 = np.eye(k)
+            PHI1 = np.eye(k)
+            
+        elif base == 'chebyshev':
+            kUse = 2*k
+            roots = Poly(chebyshevt(kUse, 2*x-1)).all_roots()
+            x_m = np.array([rt.evalf(20) for rt in roots]).astype(np.float64)
+            wm = np.pi / kUse / 2
+
+            for ki in range(k):
+                for kpi in range(k):
+                    H0[ki, kpi] = 1/np.sqrt(2) * (wm * phi[ki](x_m/2) * phi[kpi](x_m)).sum()
+                    G0[ki, kpi] = 1/np.sqrt(2) * (wm * psi(psi1, psi2, ki, x_m/2) * phi[kpi](x_m)).sum()
+                    H1[ki, kpi] = 1/np.sqrt(2) * (wm * phi[ki]((x_m+1)/2) * phi[kpi](x_m)).sum()
+                    G1[ki, kpi] = 1/np.sqrt(2) * (wm * psi(psi1, psi2, ki, (x_m+1)/2) * phi[kpi](x_m)).sum()
+
+                    PHI0[ki, kpi] = (wm * phi[ki](2*x_m) * phi[kpi](2*x_m)).sum() * 2
+                    PHI1[ki, kpi] = (wm * phi[ki](2*x_m-1) * phi[kpi](2*x_m-1)).sum() * 2
+                    
+            PHI0[np.abs(PHI0)<1e-8] = 0
+            PHI1[np.abs(PHI1)<1e-8] = 0
+
+        for matrix in [H0, H1, G0, G1]:
+            matrix[np.abs(matrix)<1e-8] = 0
+            
+        return H0, H1, G0, G1, PHI0, PHI1
```

Comment:
The function and docstring are a little hard to follow - could we add a more intuitive explanation in the doctoring and use more descriptive name for the variables (we also typically name variables e.g. `h_0` and reserve Capital Case for classes.